sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c

```diff
@@ -0,0 +1,454 @@
+  // Increment the interrupt count and detect overflows.
+  uint32_t interrupt_count =
+      flash_ctrl_testutils_counter_get(kCounterInterrupt);
+  if (interrupt_count > kMaxInterrupts) {
+    CHECK(false, "Reset count %d got too many interrupts (%d)", reset_count,
+          interrupt_count);
+  }
+  flash_ctrl_testutils_counter_set_at_least(
+      &flash_ctrl_state, kCounterInterrupt, interrupt_count + 1);
+
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+
+  LOG_INFO("Got irq_id 0x%x (%d)", irq_id, irq_id);
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralAonTimerAon) {
+    uint32_t irq =
+        (irq_id - (dif_rv_plic_irq_id_t)
+                      kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+    // We should not get aon timer interrupts since escalation suppresses them.
+    CHECK(false, "Unexpected aon timer interrupt %d", irq);
+  } else if (peripheral == kTopEarlgreyPlicPeripheralAlertHandler) {
+    CHECK(irq_id == kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+          "Unexpected irq_id, expected %d, got %d",
+          kTopEarlgreyPlicIrqIdAlertHandlerClassa, irq_id);
+  }
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
+  // register.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kPlicTarget, irq_id));
+  LOG_INFO("Regular external ISR exiting");
+}
+
+void ottf_load_integrity_error_handler(void) {
+  dif_rv_plic_irq_id_t irq_id;
+
+  LOG_INFO("At load integrity error handler");
+
+  uint32_t mtval = ibex_mtval_read();
+  CHECK(mtval == kErrorRamAddress, "Unexpected mtval: expected 0x%x, got 0x%x",
+        kErrorRamAddress, mtval);
+
+  rv_core_ibex_fault_checker(true);
+
+  // There may be multiple interrupts due to the alert firing, so this keeps an
+  // interrupt counter and errors-out if there are too many interrupts.
+
+  // Increment the interrupt count and detect overflows.
+  uint32_t interrupt_count =
+      flash_ctrl_testutils_counter_get(kCounterInterrupt);
+  if (interrupt_count > kMaxInterrupts) {
+    CHECK(false, "Reset count %d got too many interrupts (%d)", reset_count,
+          interrupt_count);
+  }
+  flash_ctrl_testutils_counter_set_at_least(
+      &flash_ctrl_state, kCounterInterrupt, interrupt_count + 1);
+
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+
+  LOG_INFO("Got irq_id 0x%x (%d)", irq_id, irq_id);
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralAonTimerAon) {
+    uint32_t irq =
+        (irq_id - (dif_rv_plic_irq_id_t)
+                      kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+    // We should not get aon timer interrupts since escalation suppresses them.
+    CHECK(false, "Unexpected aon timer interrupt %d", irq);
```

Comment:
also I don't think you need this check here do? it makes sense in the `external_isr` handler, but the aon timer irq should never be the cause of the this synchronous exception handler, no? are you trying to test if these events magically occur at the same time?

Reply:
I agree it is redundant. I'll clean it up in a subsequent PR.